trunk/source/BuchiProgramProduct/src/de/uni_freiburg/informatik/ultimate/buchiprogramproduct/productgenerator/ProductGenerator.java

```diff
@@ -453,23 +465,20 @@ private void pruneNonProductSinks() {
 		// all helper states that have edges to non-product states receive a
 		// self-loop that keeps them in the same product state as the LTL NWA
 
-		for (final BoogieIcfgLocation helper : mHelperProductStates) {
+		for (final BoogieIcfgLocation helper : new HashSet<>(mHelperProductStates)) {

```

Comment:
does `pruneNonProductSink` generate new helper product states or why this `HashSet` here?